pkg/local-gadget-manager/local-gadget-manager.go

```diff
@@ -302,6 +304,24 @@ func NewManager() (*LocalGadgetManager, error) {
 		return nil, err
 	}
 
+	// Ensure /sys/fs/bpf is of type bpf. It is necessary to be able to pin eBPF
+	// maps. TODO: Remove the need of using pinning, see issues #619 and #620.
+	const bpfPinPath string = "/sys/fs/bpf"
+	var buf unix.Statfs_t
+	if err := unix.Statfs(bpfPinPath, &buf); err != nil {
+		return nil, fmt.Errorf("error checking type of %s: %w", bpfPinPath, err)
+	}
+	if buf.Type != unix.BPF_FS_MAGIC {
+		log.Debugf("%s is of type %#x but we were expecting %#x (BPF_FS_MAGIC)",
+			bpfPinPath, buf.Type, unix.BPF_FS_MAGIC)
+
+		log.Infof("Remounting %q as bpf", bpfPinPath)
+
+		if err := unix.Mount("bpf", bpfPinPath, "bpf", 0, ""); err != nil {
+			return nil, fmt.Errorf("error remounting %s as bpf: %w", bpfPinPath, err)
+		}
+	}
+
```

Comment:
Could this be moved in a separate function, even if it is in the same file for now?
```
func ensureBPFMount() error
```